src/tribler/gui/network/request/request.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import annotations
+
+import json
+import logging
+from time import time
+from typing import Callable, Dict, Optional, TYPE_CHECKING, Union
+
+from PyQt5.QtCore import QObject, pyqtSignal
+from PyQt5.QtNetwork import QNetworkReply, QNetworkRequest
+
+from tribler.gui.utilities import connect
+
+if TYPE_CHECKING:
+    from tribler.gui.network.request_manager import RequestManager
+
+
+class Request(QObject):
+    GET = 'GET'
+    POST = 'POST'
+    PUT = 'PUT'
+    PATCH = 'PATCH'
+    DELETE = 'DELETE'
+
+    # This signal is called if we receive some real reply from the request
+    # and if the user defined a callback to call on the received data.
+    # We implement the callback as a signal call and not as a direct callback
+    # because we want the request object be deleted independent of what happens
+    # during the callback call.
+    on_finished_signal = pyqtSignal(object)
+    on_cancel_signal = pyqtSignal()
+
+    def __init__(
+            self,
+            endpoint: str,
+            on_finish: Callable = lambda _: None,
+            on_cancel: Callable = lambda: None,
+            url_params: Optional[Dict] = None,
+            data: Optional[Union[bytes, str, Dict]] = None,
+            method: str = GET,
+            capture_errors: bool = True,
+            priority=QNetworkRequest.NormalPriority,
+            raw_response: bool = False,
+    ):
+        super().__init__()
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+        self.endpoint = endpoint
+        self.url_params = url_params
+
+        self.priority = priority
+        self.method = method
+        self.capture_errors = capture_errors
+        self.raw_response = raw_response
+        self.data: Optional[bytes] = data
+        self.raw_data = data
+        if isinstance(self.data, Dict):
+            self.data = json.dumps(data)
+        if isinstance(self.data, str):
+            self.data = self.data.encode('utf8')
```

Comment:
Here, `raw_data` is the original data provided with `Request`, and `data` is an encoded version of it.

Usually, the meaning of `data` and `raw_data` is the other way around. For example, this is what ChatGPT thinks about it:

> If a Request object has attributes `data` and `raw_data`, what can be the value of these attributes?

> The data attribute of a Request object is typically used to store parsed and processed data that has been extracted from the request, such as form data or JSON data. The raw_data attribute, on the other hand, would likely store the original, raw data that was sent in the request, such as the bytes of a file or a string of XML or JSON. The exact values that these attributes can hold would depend on the implementation of the Request object.

Therefore, I suggest swapping these attributes to avoid confusion for future programmers.